butane_core/src/migrations/adb.rs

```diff
@@ -369,6 +430,14 @@ impl AColumn {
     pub fn default(&self) -> &Option<SqlVal> {
         &self.default
     }
+    /// Returns whether this column refers to another column.
+    pub fn reference(&self) -> &Option<ARef> {
+        &self.reference
+    }
+    /// Set another column that this column refers to.
+    pub fn add_reference(&mut self, reference: &ARef) {
```

Comment:
I don't follow why this method needs to exist. The only usage I see (in `butane_core/codegen/migration.rs`) is at the same time we're constructing an `AColumn`. So why isn't the constructor parameter sufficient?

Reply:
Using `AColumn::new_simple` with an optional `::add_reference` is a bit easier to read, but I am happy to switch to using two different invocations of `::new`

Reply:
I'd rather not have the extra method because the column isn't really supposed to be changed after it's created. But I agree with the readability issue, so I won't block here.
The solution to both aspects is to switch to a builder pattern, and I can do that in a follow-up.